pkg/storkctl/migrationschedule.go

```diff
@@ -43,22 +48,77 @@ func newCreateMigrationScheduleCommand(cmdFactory Factory, ioStreams genericclio
+			migrationScheduleNs := cmdFactory.GetNamespace()
+			clusterPairObj, err := storkops.Instance().GetClusterPair(clusterPair, migrationScheduleNs)
+			if err != nil {
+				util.CheckErr(fmt.Errorf("unable to find the cluster pair in the given namespace"))
+				return
+			}
+			isSyncDr := false
+			//clusterPairObj.Spec.Options map is empty for syncDr and non-empty for asyncDR
+			if len(clusterPairObj.Spec.Options) == 0 {
+				isSyncDr = true
+			}
+
+			// Default value of includeVolumes for syncDr is false and for async DR is true
+			if isSyncDr {
+				if c.Flags().Changed("includeVolumes") && includeVolumes {
+					util.CheckErr(fmt.Errorf("IncludeVolumes can only be set to false because the volumes are already present " +
+						"on the destination cluster as we deal with a single Portworx cluster in sync-dr usecases"))
+					return
+				}
+				includeVolumes = false
+			}
+
 			if len(namespaceList) == 0 {
 				util.CheckErr(fmt.Errorf("need to provide atleast one namespace to migrate"))
 				return
 			}
-			if len(schedulePolicyName) == 0 {
-				util.CheckErr(fmt.Errorf("need to provide schedulePolicyName"))
+
+			//Allow the user to only add one of the flags schedulePolicyName or interval
+			if c.Flags().Changed("schedulePolicyName") && c.Flags().Changed("interval") {
+				util.CheckErr(fmt.Errorf("must provide only one of schedulePolicyName or interval values"))
 				return
 			}
 
-			_, err := storkops.Instance().GetSchedulePolicy(schedulePolicyName)
-			if err != nil {
-				util.CheckErr(fmt.Errorf("error getting schedulepolicy %v: %v", schedulePolicyName, err))
-				return
+			//If value for interval is explicitly set by the user, we will ignore schedulePolicyName
+			if c.Flags().Changed("interval") {
+				var policyItem storkv1.SchedulePolicyItem
+				var intervalPolicy storkv1.IntervalPolicy
+				intervalPolicy.IntervalMinutes = intervalMinutes
+				intervalPolicy.Retain = storkv1.DefaultIntervalPolicyRetain
+				// Validate the user input
+				err := intervalPolicy.Validate()
+				if err != nil {
+					util.CheckErr(fmt.Errorf("could not create a schedule policy with specified interval. %v", err))
+					return
+				}
+				policyItem.Interval = &intervalPolicy
+				//name of the schedule policy created will be same as the migration schedule name provided as an input.
+				schedulePolicyName = migrationScheduleName
+				var schedulePolicy storkv1.SchedulePolicy
+				schedulePolicy.Name = schedulePolicyName
+				schedulePolicy.Policy = policyItem
+				_, err = storkops.Instance().CreateSchedulePolicy(&schedulePolicy)
+				if err != nil {
+					util.CheckErr(fmt.Errorf("could not create a schedule policy with specified interval. %v", err))
+					return
+				}
+			} else {
+				//we reach here if user has not set the interval flag value.
+				//So we take either the user input for schedulePolicyName or the default schedulePolicyName
+				logrus.Infof("schedulePolicy name is %s", schedulePolicyName)
```

Comment:
I didn't understand the comment here.. "or the default schedulePolicyName". Can you explain?

Reply:
There are 4 cases in all
1. user doesn't provide schedulePolicyName nor interval value -> we should go ahead with the "default-migration-policy"
2. user provides both schedulePolicyName and interval value -> we throw an error saying you can only fill one of the values
3. user provides interval value only -> we will create a new schedule policy and use that in the migrationschedule
4. user provides schedulePolicyName only -> we will check if such a schedule policy exists and if yes use that schedulePolicy.

So in my code case 1 and 4 are being covered in this branch

Reply:
Have added better comments to make this code block self-explanatory.